src/nfa/nfa.cc

```diff
@@ -74,6 +74,70 @@ namespace {
         LTSforSimulation.init();
         return LTSforSimulation.compute_simulation();
     }
+
+	void reduce_size_by_simulation(Nfa* result, const Nfa& aut, StateMap<State> &state_map) {
+        auto sim_relation = compute_relation(aut, StringDict{{"relation", "simulation"}, {"direction","forward"}});
+
+        auto sim_relation_symmetric = sim_relation;
+        sim_relation_symmetric.restrict_to_symmetric();
+
+        // for State q, quot_proj[q] should be the representative state representing the symmetric class of states in simulation
+        std::vector<size_t> quot_proj;
+        sim_relation_symmetric.get_quotient_projection(quot_proj);
+
+		// map each state q of aut to the state of the reduced automaton representing the simulation class of q
+		for (State q = 0; q < aut.get_num_of_states(); ++q) {
+			State qReprState = quot_proj[q];
+			if (state_map.count(qReprState) == 0) { // we need to map q's class to a new state in reducedAut
+				State qClass = result->add_new_state();
+				state_map[qReprState] = qClass;
+				state_map[q] = qClass;
+			} else {
+				state_map[q] = state_map[qReprState];
+			}
+		}
+
+        for (State q = 0; q < aut.get_num_of_states(); ++q) {
```

Comment:
Joint comment also for line 89: Maybe it would be good to extract the call to the `get_num_of_states()`: 1. if it does not change during two calls, it could save some computation time, and second it is vulnerable to potential accidental quadratic complexity.

Reply:
I extracted the call to the `get_num_of_states()`, check if it is all right now